### PR TITLE
Avoid spurious warnings from the usb serial plugin & clean up examples

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

File: examples/ShowRecorder.cpp [hunk 1]
```diff
@@ -104,22 +105,19 @@ void ShowRecorder::Stop() {
 /**
  * Record the new frame
  */
-void ShowRecorder::NewFrame(unsigned int universe,
-                            const DmxBuffer &data,
-                            const string &error) {
-  if (!error.empty()) {
```
Comment (peternewman) [inline examples/ShowRecorder.cpp]: How does the new client report errors, should you be trapping something else?

File: examples/ShowRecorder.cpp [hunk 2]
```diff
@@ -104,22 +105,19 @@ void ShowRecorder::Stop() {
...
 
 
-void ShowRecorder::RegisterComplete(const string &error) {
-  OLA_INFO << "register complete";
-  (void) error;
+void ShowRecorder::RegisterComplete(const Result &result) {
+  if (!result.Success()) {
+    OLA_WARN << "Register failed: " << result.Error();
+  } else {
+    OLA_INFO << "register complete";
```
Comment (peternewman) [inline examples/ShowRecorder.cpp]: SPaG "Register completed" please

File: examples/ola-client.cpp [hunk 3]
```diff
@@ -852,11 +859,8 @@ int SendDmx(OlaCallbackClientWrapper *wrapper, const options &opts) {
     exit(1);
   }
 
-  if (!client->SendDmx(opts.uni, buffer,
-                       NewSingleCallback(&SendDmxComplete, ss))) {
-    cout << "Send DMX failed" << endl;
-    return 1;
-  }
+  ola::client::SendDMXArgs args(NewSingleCallback(&SendDmxComplete, ss));
+  client->SendDMX(opts.uni, buffer, args);
```
Comment (peternewman) [inline examples/ola-client.cpp]: Does this not return a state to say if sending failed?

Comment (peternewman) [inline examples/ola-client.cpp]: Or is this covered in the SendDmxComplete CB?

File: examples/ola-dmxmonitor.cpp [hunk 4]
```diff
@@ -240,14 +239,13 @@ void DmxMonitor::NewDmx(unsigned int universe,
   clock.CurrentTime(&m_last_data);
   Values();
   refresh();
-  (void) universe;
-  (void) error;
+  (void) meta;
```
Comment (peternewman) [inline examples/ola-dmxmonitor.cpp]: OLA_UNUSED?

File: include/ola/io/IOUtils.h [hunk 5]
```diff
@@ -37,6 +37,17 @@ namespace io {
...
 bool Open(const std::string &path, int oflag, int *fd);
+
+/**
+ * @brief Wrapper around open().
+ * @param path the path to open
+ * @param oflag flags passed to open
+ * @param[out] fd a pointer to the fd which is returned.
+ * @returns true if the open succeeded, false otherwise.
+ *
+ * This is similar to Open(), but doesn't log a warning if open() fails.
```
Comment (peternewman) [inline include/ola/io/IOUtils.h]: Can we move this below @brief, then the @param after it please.

Comment (peternewman) [inline include/ola/io/IOUtils.h]: @seealso on this and/or Open()

**Comment by peternewman** [comment]
Looks good, just minor comments.

File: examples/ShowRecorder.cpp (hunk 1 above)
Comment (nomis52) [inline examples/ShowRecorder.cpp]: It doesn't. If an error occurs it simply doesn't call the handler.

File: examples/ShowRecorder.cpp (hunk 2 above)
Comment (nomis52) [inline examples/ShowRecorder.cpp]: Done

File: examples/ola-client.cpp (hunk 3 above)
Comment (nomis52) [inline examples/ola-client.cpp]: Yes, the callback will be run if the send fails.

File: examples/ola-dmxmonitor.cpp (hunk 4 above)
Comment (nomis52) [inline examples/ola-dmxmonitor.cpp]: Done

File: include/ola/io/IOUtils.h (hunk 5 above)
Comment (nomis52) [inline include/ola/io/IOUtils.h]: Done

**Pull request merged**